block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockMetadataStore.scala

```diff
@@ -41,11 +41,22 @@ object BlockMetadataStore {
       dagSet: Set[BlockHash],
       childMap: Map[BlockHash, Set[BlockHash]],
       heightMap: SortedMap[Long, Set[BlockHash]],
+      // In general - at least genesis should be LFB.
+      // But dagstate can be empty, as it is initialized before genesis is inserted.
+      // Also lots of tests do not have genesis properly initialised, so fixing all this is pain.
+      // So this is Option.
+      lastFinalizedBlock: Option[BlockHash],
```

Comment:
LFB is option here because there should be empty DagState as init value to recreate inMem representation.